cmd/package-operator-manager/main.go

```diff
@@ -77,13 +92,91 @@ func main() {
 		os.Exit(2)
 	}
 
-	if err := run(setupLog, scheme, opts); err != nil {
+	if len(opts.loadPackage) > 0 {
+		namespace, name, found := strings.Cut(opts.loadPackage, string(types.Separator))
+		if !found {
+			fmt.Fprintln(os.Stderr, "invalid argument to --load-package, expected NamespaceName")
+			os.Exit(1)
+		}
+
+		packageKey := client.ObjectKey{
+			Name:      name,
+			Namespace: namespace,
+		}
+		if err := runLoader(scheme, packageKey); err != nil {
+			setupLog.Error(err, "unable to run package-loader")
+			os.Exit(1)
+		}
+		return
+	}
+
+	if len(opts.copyTo) > 0 {
+		if err := runCopyTo(opts.copyTo); err != nil {
+			setupLog.Error(err, "unable to run copy-to")
+			os.Exit(1)
+		}
+		return
+	}
+
+	if err := runManager(setupLog, scheme, opts); err != nil {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 }
 
-func run(log logr.Logger, scheme *runtime.Scheme, opts opts) error {
+func runCopyTo(target string) error {
+	src, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("looking up current executable path: %w", err)
+	}
+
+	srcFile, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("opening source file: %w", err)
+	}
+	defer srcFile.Close()
+
+	destFile, err := os.Create(target)
+	if err != nil {
+		return fmt.Errorf("opening destination file: %w", err)
+	}
+	defer destFile.Close()
+
+	_, err = io.Copy(destFile, srcFile)
+	if err != nil {
+		return fmt.Errorf("copy: %w", err)
+	}
+
+	return os.Chmod(destFile.Name(), 0755)
+}
+
+const packageFolderPath = "/package"
+
+func runLoader(scheme *runtime.Scheme, packageKey client.ObjectKey) error {
+	c, err := client.New(ctrl.GetConfigOrDie(), client.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		return fmt.Errorf("creating target cluster client: %w", err)
```

Comment:
Is this really the target cluster client? Like the hosted cluster?

Reply:
My bad, copy/pasta issue :)
Should be fixed now!